src/Tools/dotnet-monitor/CollectionRules/ExecuteAction.cs

```diff
@@ -0,0 +1,33 @@
+﻿using Microsoft.Diagnostics.Tools.Monitor.CollectionRules.Options.Actions;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Diagnostics.Tools.Monitor.CollectionRules
+{
+    internal sealed class ExecuteAction : IAction<ExecuteOptions>
+    {
+        public async Task<ActionResponse> Execute(ExecuteOptions options, CancellationToken token)
+        {
+            ActionResponse executeResponse = new();
+
+            await Task.Run(() =>
+            {
+                string path = options.Path;
+                string arguments = options.Arguments;
+
+                Process process = Process.Start(path, arguments);
+
+                process.WaitForExit();
+
+                int exitCode = process.ExitCode;
```

Comment:
A non-zero exit code should fail the action. Maybe we need another property (e.g. `IgnoreExitCode`) on ExecuteOptions class to state that the exit code should be ignored, but that behavior should be opt-in.

Reply:
When you say "should fail the action," what did you have in mind? In the current state, the `ExitCode` could be processed by the caller, but if we don't want to enforce that responsibility on it, would we want to throw from `ExecuteAsync`? Or something else entirely? And in the event that a user has set `IgnoreExitCode` and another action is dependent on the result of the current action, what behavior should we expect?

Reply:
> When you say "should fail the action," what did you have in mind?

The action should throw an exception with a specific message e.g. `The process exited with exit code {0}`.

> In the current state, the `ExitCode` could be processed by the caller, but if we don't want to enforce that responsibility on it, would we want to throw from `ExecuteAsync`?

The caller should not have an specific knowledge of any of the properties in the result.

> And in the event that a user has set `IgnoreExitCode` and another action is dependent on the result of the current action, what behavior should we expect?

I would expect that it still fills out the same properties in the result, but they might be null, string.empty, or default(T) depending on the type. So if this action is reporting `ExitCode`, `StandardOut`, and `StandardError` when the process normally exits, I'd still expect it to write out those properties; you'll still have an exit code, standard out, and standard error for a process with a non-zero exit code.